### DISCUSSION TITLE
Deploying Hasura on Kubernetes has "database query error"

### DISCUSSION THREAD

**Discussion by tz-torchai** [hasura/graphql-engine#6721] [Question]
Hi, when i first installed hasura using helm, it worked well: it reads the databagse connection string from env and connects to it automatically. 

however when i reinstall it, it does not connect to the database automatically anymore. then when i try to add the database manually via hasura console. it failed with error:
```
Add data source failed
database query error
    {
        "definition": "yugabyte",
        "reason": "database query error",
        "type": "source"
    }
```

The log in hasura shows: 
```
{"type":"http-log","timestamp":"2021-03-28T08:50:10.910+0000","level":"error","detail":{"operation":{"user_vars":{"x-hasura-role":"admin"},"error":{"internal":[{"definition":"yugabyte","reason":"database query error","type":"source"}],"path":"$.args","error":"database query error","code":"constraint-violation"},"request_id":"47638505-5de4-44d9-bd6c-66d5a9d889a3","response_size":165,"query":{"resource_version":null,"args":{"name":"yugabyte","configuration":{"connection_info":{"database_url":{"from_env":"HASURA_GRAPHQL_DATABASE_URL"},"pool_settings":{}}}},"version":1,"type":"pg_add_source"}},"request_id":"47638505-5de4-44d9-bd6c-66d5a9d889a3","http_info":{"status":400,"http_version":"HTTP/1.1","url":"/v1/metadata","ip":"2001:1970:5b99:1100:7839:9de3:8639:2cbd","method":"POST","content_encoding":null}}}
```

I can isolate the error is not caused by the database cuz i can connect to it by other means.

What could be wrong here? Let me know if there is any other information I can provide. Thanks!

**Comment by ealtuna** [comment]
I am having the same error using the docker-compose deployment with v2.0.0-alpha.6. I think it will be great to have additional information that helps the user figure out what is the concrete error.
